Modules/ITS/src/ITSTrackTask.cxx

```diff
@@ -0,0 +1,222 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// // distributed under the terms of the GNU General Public License v3 (GPL
+// // Version 3), copied verbatim in the file "COPYING".
+// //
+// // See http://alice-o2.web.cern.ch/license for full licensing information.
+// //
+// // In applying this license CERN does not waive the privileges and immunities
+// // granted to it by virtue of its status as an Intergovernmental Organization
+// // or submit itself to any jurisdiction.
+//
+
+///
+/// \file   ITSTrackTask.cxx
+/// \author Artem Isakov
+///
+
+#include "QualityControl/QcInfoLogger.h"
+#include "ITS/ITSTrackTask.h"
+#include <DetectorsCommonDataFormats/NameConf.h>
+#include <DataFormatsITS/TrackITS.h>
+#include <DataFormatsITSMFT/ROFRecord.h>
+
+using namespace o2::itsmft;
+using namespace o2::its;
+
+namespace o2::quality_control_modules::its
+{
+
+ITSTrackTask::ITSTrackTask() : TaskInterface()
+{
+
+  getEnableLayers();
+  createAllHistos();
+
+  o2::base::GeometryManager::loadGeometry();
+  mGeom = o2::its::GeometryTGeo::Instance();
+}
+
+ITSTrackTask::~ITSTrackTask()
+{
+
+  delete hNClusters;
+  delete hTrackEta;
+  delete hTrackPhi;
+  delete hOccupancyROF;
+  delete hClusterUsage;
+  delete hAngularDistribution;
+
+  delete mGeom;
+}
+
+void ITSTrackTask::initialize(o2::framework::InitContext& /*ctx*/)
+{
+
+  QcInfoLogger::GetInstance() << "initialize ITSTrackTask" << AliceO2::InfoLogger::InfoLogger::endm;
+  publishHistos();
+  std::string dictfile = o2::base::NameConf::getDictionaryFileName(o2::detectors::DetID::ITS, "", ".bin");
```

Comment:
how will you guarantee that you have this file in production ? or will you get this dictionary from the CCDB in production ?

Reply:
The dictionary will be created by the ITS reconstruction device. I have no idea how to get it in production at the moment but will confirm that with reconstruction people.

Reply:
Ok, it can stay like that for now but perhaps add a "todo" or a ticket in JIRA to remember to update it to something else for production. You could perhaps use the CCDB ?

Reply:
I guess so. I will discuss it with Jouri. The ticket was created: https://alice.its.cern.ch/jira/browse/QC-467
Thank you.